File: 백준/Silver/1929. 소수 구하기/소수 구하기.java
import java.util.*;
import java.io.*;
public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int m = Integer.parseInt(st.nextToken());
        int n = Integer.parseInt(st.nextToken());

        int [] array = new int [n+1];

        for(int i=2; i<=n; i++){
            array[i] = i;
        }

        for(int i=2; i<=Math.sqrt(n); i++){

            if (array[i]==0){
                continue;
            }

            for(int j=i*2; j<=n; j+=i){
                array[j]=0;
            }

        }

        for(int i=m; i<=n; i++){
            if(array[i]!=0){
                System.out.println(array[i]);
            }
        }

    }
}

File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
# 소수 구하기
import sys
import math
input = sys.stdin.readline
m, n = map(int, input().split())
field=[True]*(n+1)
field[0]=False
field[1]=False

for i in range(2, int(math.sqrt(n))+1):
    if field[i]==False:
            continue
    for j in range(2*i, n+1, i):
        field[j] = False

for i in range(m, n+1):
    if field[i]==True:
        print(i)